source/operations/install-deploy-manage/deploy-operator-kustomize.rst

```diff
@@ -0,0 +1,218 @@
+.. _minio-k8s-deploy-operator-kustomize:
+
+==============================
+Deploy Operator With Kustomize
+==============================
+
+.. default-domain:: minio
+
+.. contents:: Table of Contents
+   :local:
+   :depth: 2
+
+
+Overview
+--------
+
+`Kustomize <https://kubernetes.io/docs/tasks/manage-kubernetes-objects/kustomization>`__ is a YAML-based templating tool that allows you to define Operator and Tenant templates using plain Kubernetes YAML reference language.
+Kustomize is included with the `Kubectl <https://kubernetes.io/docs/reference/kubectl/>`__ command line tool.
+
+The `default MinIO Operator template <https://github.com/minio/operator/blob/master/kustomization.yaml>`__ provides a starting point for customizing configurations for your local environment.
+You can modify a Kustomize Operator deployment after installation with patches, which are additional YAML configurations applied over the existing base.
+
+
+Prerequisites
+-------------
+
+To install the Operator with Kustomize you will need the following:
+
+* An existing Kubernetes cluster, v1.21 or later.
+* The ``kubectl`` CLI tool on your local host, the same version as the cluster.
+* `Docker <https://docker.com>`__ for your platform.
+* Access to run ``kubectl`` commands on the cluster from your local host.
+
+For more about Operator installation requirements, including TLS certificates, see the :ref:`Operator deployment prerequisites <minio-operator-prerequisites>`.
+
+This procedure assumes familiarity with the referenced Kubernetes concepts and utilities.
+While this documentation may provide guidance for configuring or deploying Kubernetes-related resources on a best-effort basis, it is not a replacement for the official :kube-docs:`Kubernetes Documentation <>`.
+
+.. _minio-k8s-deploy-operator-kustomize-repo:
+
+Install the MinIO Operator using Kustomize
+------------------------------------------
+
+The following procedure uses ``kubectl --kustomize`` to install the Operator from the MinIO Operator GitHub repository:
+
+.. important::
+
+   Do not use ``kubectl krew`` or similar methods to update or manage the MinIO Operator installation.
+   If you use Kustomize to install the Operator, you must use Kustomize to manage that installation.
+
+#. Install the latest version of Operator
+
+   .. code-block:: shell
+      :class: copyable
+      :substitutions:
+
+      kubectl apply -k github.com/minio/operator\?ref=|operator-version-stable|
```

Comment:
Sometimes we've gotten this warning while testing, although the command seems to succeed after that. What causes it? Is this something we need to cover?
```
kubectl apply -k github.com/minio/operator\?ref=v5.0.14
namespace/minio-operator created
Warning: resource customresourcedefinitions/miniojobs.job.min.io is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
customresourcedefinition.apiextensions.k8s.io/miniojobs.job.min.io configured
```

Reply:
Need to sort out order of operations here. This is probably because no PV exists yet?